### PR TITLE
feat(mwpw-155425): ability to have multiple event sort options

### PR DESCRIPTION
**Pull request by sheridansunier**

### CODE REVIEW

**Review by sanrai — CHANGES_REQUESTED** [review]
I think there may be a logic error with this code. 

Can you check these test cases (you can copy/paste snippet into chrome console) and let me know if this is expected behavior?

If it is a bug, we should probably write a unit test for this as it was tricky to catch.

```
// Sample data representing different event categories
var live = ['live1', 'live2'];
var upComing = ['upcoming1', 'upcoming2'];
var onDemand = ['ondemand1', 'ondemand2'];
var notTimed = ['nottimed1', 'nottimed2'];

// Buggy version of the eventTiming function
function getCardsBuggy(sanitizedEventFilter) {
    let cards = [];
    if (sanitizedEventFilter.length === 0) {
        cards = [].concat(live, upComing, onDemand, notTimed);
    } 
    if (sanitizedEventFilter.indexOf('live') > -1) {
        cards = cards.concat(live);
    } 
    if (sanitizedEventFilter.indexOf('upcoming') > -1) {
        cards = cards.concat(upComing);
    } 
    if (sanitizedEventFilter.indexOf('on-demand') > -1) {
        cards = cards.concat(onDemand);
    } 
    if (sanitizedEventFilter.indexOf('not-timed') > -1) {
        cards = cards.concat(notTimed);
    } else {
        cards = [].concat(live, upComing, onDemand, notTimed);
    }
    return cards;
}

// Corrected version of the eventTiming function
function getCardsCorrected(sanitizedEventFilter) {
    let cards = [];
    if (sanitizedEventFilter.length === 0) {
        cards = [].concat(live, upComing, onDemand, notTimed);
    } else {
        if (sanitizedEventFilter.indexOf('live') > -1) {
            cards = cards.concat(live);
        }
        if (sanitizedEventFilter.indexOf('upcoming') > -1) {
            cards = cards.concat(upComing);
        }
        if (sanitizedEventFilter.indexOf('on-demand') > -1) {
            cards = cards.concat(onDemand);
        }
        if (sanitizedEventFilter.indexOf('not-timed') > -1) {
            cards = cards.concat(notTimed);
        }
    }
    return cards;
}

// Test cases to compare the buggy and corrected functions
var testFilters = [
    [],
    ['live'],
    ['upcoming'],
    ['on-demand'],
    ['not-timed'],
    ['live', 'on-demand'],
    ['invalid-filter']
];

// Function to run the tests and log the results
function runTests() {
    testFilters.forEach(filter => {
        console.log(`Testing with filter: ${JSON.stringify(filter)}`);

        // Using the buggy function
        var buggyResult = getCardsBuggy(filter);
        console.log('Buggy Function Result:', buggyResult);

        // Using the corrected function
        var correctedResult = getCardsCorrected(filter);
        console.log('Corrected Function Result:', correctedResult);

        console.log('-------------------------');
    });
}

// Run the tests
runTests();
```

**Comment by sanrai** [comment]
Hi @sheridansunier,

Please address comments in PR.

Below is a summary of what I tested with this PR.

## Summary of Tested Functionality
### 1. Multi-Event Filter Functionality
- **Support for multiple event states**:  
  - Verified that authors can select *any* combination of `[live, upcoming, on-demand, not-timed]` or `"all"`.
  - Confirmed that `sanitizeEventFilter` and the updated `eventTiming` logic return the correct subset of cards (e.g., `["live", "on-demand"]` shows exactly live + on-demand, etc.).
  - Ensured `"all"` or an *empty* array means “include all categories” in the final card set.

- **Individual states** tested:
  - **Live** (start < now < end)
  - **Upcoming** (start > now)
  - **On-demand** (end < now)
  - **Not-timed** (missing or zero start/end date)
  - Verified each bucket independently with single-string or array filters.

- **Concatenation order**:
  - Empty filter => all categories in order `[live, upcoming, on-demand, not-timed]`.
  - Multi-state filter => merges only those buckets in ascending start date order (e.g., `["live", "upcoming"]` merges those two categories).

---

### 2. `nextTransitionMs` Logic
- **Purpose**: tracks the earliest moment (in ms) at which any card in the filtered set transitions to a new state (e.g., live → on-demand at endDate, or upcoming → live at startDate).
- **Multiple transitions**:
  - Tested multiple *live* events ending at different times + multiple *upcoming* events starting at different times.
  - Confirmed `nextTransitionMs` consistently reflects the **earliest** transition among them.
  - Returns `0` if no transitions are pending (e.g., no live or upcoming cards).

---

### 3. Sort and Bucket Behavior
- **Sorting**:  
  - Verified sessions are initially sorted by ascending `startDate`.
  - Cards without valid dates (i.e., “not-timed”) go at the end.

- **Bucketing**:  
  - Confirmed the logic that categorizes cards into live, upcoming, on-demand, or not-timed.
  - Includes checks for tagged exceptions (e.g., `on-demand-scheduled` or `live-expired`).



**In short**:  
- I verified **new multi-filter event states** work correctly.  
- **Sorting** and **bucketing** logic is intact.  
- **`nextTransitionMs`** accurately picks the soonest transition time among multiple live/upcoming events.  

This covers the scope of the PR and confirms it meets the requirements for multi-event selection while preserving the correct time-based transitions.

File: react/src/js/components/Consonant/Container/Container.jsx [hunk 1]
```diff
@@ -169,6 +170,8 @@ const Container = (props) => {
     const categories = getConfig('filterPanel', 'categories');
     // eslint-disable-next-line no-use-before-define, max-len
     const authoredCategories = isCategoriesContainer ? getAuthoredCategories(authoredFilters, categories) : [];
+    // eslint-disable-next-line no-use-before-define
```
[the remark this answers is not in the document]
Comment (sheridansunier) [reply react/src/js/components/Consonant/Container/Container.jsx:173]: not anymore - it was earlier lol

File: react/src/js/components/Consonant/Helpers/Helpers.js [hunk 2]
```diff
@@ -413,29 +413,30 @@ export const getDateDescSort = cards => getDateAscSort(cards).reverse();
...
+        overlays: card.overlays || {},
+        showCard: card.showCard || {},
+        search: card.search || {},
+        styles: card.styles || {},
     }));
 
     const result = eventTiming(transformedCards, eventFilter);
 
-    const visibleSessions = result.visibleSessions
-        .filter(session => session.tags.includes(eventFilter))
```
Comment (sanrai) [inline react/src/js/components/Consonant/Helpers/Helpers.js:424]: A side effect of this is that if certain cards relied on being tagged (instead of date/time) to be displayed, they might no longer appear.

Previously, the code used `tags.includes("live" || "on-demand" || "upcoming")` to decide which card appeared. 

Now this change makes it so it only looks at the actual `startTime/endTime` to determine if an event is `live`, `upcoming`, or `on-demand`.

**Review by sanrai — APPROVED** [review]
